doc/whats_new/v1.0.rst

```diff
@@ -134,6 +134,11 @@ Changelog
   class methods and will be removed in 1.2.
   :pr:`18543` by `Guillaume Lemaitre`_.
 
+- |Feature| :func:`metrics.pinball_loss` exposes the pinball loss for
+  quantile regression. :class:`ensemble.GradientBoostingRegressor`
+  uses such loss to fit a quantile regression. :pr:`19415`
```

Comment:
I would not mention GradientBoostingRegressor in the changelog because this class implementation was not impacted by the PR.

```suggestion
  quantile regression. :pr:`19415`
```

I think it will be good to update the doc with the list of scikit-learn estimators that can directly be trained by using this loss as internal objective function but for now as we only have GradientBoostingRegressor this would feel weird and it's only mentioned in the inline code snippets of the user guide.

The user guide should be updated once PRs linked from #18997 are merged.